### PR TITLE
Add CircleCI, fixes #19

### LINKED ISSUE #19: Add CI
I really love CircleCI and it is very easy to setup.

PS: actually, will PR

### PR DESCRIPTION
**Pull request by nkbt**
Run lint and tests in CI

The only thing I cannot do - enable CircleCI for the main repo. I did it for my fork, but one of maintainers need to enable it for the main one.

### CODE REVIEW

**Comment by nkbt** [comment]
https://circleci.com/gh/nkbt/npm-http-server/3 this run. When main repo is enabled, CircleCI will give build notifications automatically for each commit and PR.

File: modules/parsePackageURL.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { parse as parseURL } from 'url'
 const PathnameFormat = /^\/((?:@[^\/@]+\/)?[^\/@]+)(?:@([^\/]+))?(\/.+)?$/
 
 function decodeParam(param) {
-  return param && decodeURIComponent(param)
+  return param ? decodeURIComponent(param) : null
```
Comment (nkbt) [inline modules/parsePackageURL.js]: RegExp exec on line 11 returns `undefined` and not `null` for non-matched parts, so tests were failing.

**Comment by nkbt** [comment]
Fixes #19

**Comment by mjackson** [comment]
I usually use Travis CI, but I've been meaning to try Circle CI. Do you like Circle CI more? Why?

**Comment by kentcdodds** [comment]
If you're interested, I spent some time reviewing several CI services and blogged about it: https://medium.com/continuous-delivery/continuous-delivery-3a4a55baa58a

Personally, I think that because this project isn't a library but an actual product, Snap would lend itself pretty well to this. Just my 2￠

**Comment by mjackson** [comment]
This is going to sound really dumb, but I just don't want to log in to yet another CI provider. :P

**Comment by kentcdodds** [comment]
I don't think that's dumb at all. I'm a big fan of [PDD](http://www.markhneedham.com/blog/2011/08/21/pain-driven-development/) :+1:

**Comment by nkbt** [comment]
Killer features for me: lightning fast and ssh to a docker box (dynamic for each new box created for every run).

We use CircleCi for about a year for our main big-data product (about 50 microservices) and I personally use it for all my libs and projects (another 20 I reckon).

Since we do a lot of bash scripting, auto-deployment, docker mangling it is extremely helpful when something does not work as expected - ssh to a test-box (for particular run) and reproduce issue manually. OSX is close to Linux, but still different in many parts.

And I used Travis, Jenkins too a lot, so must say that CircleCI is far more superior.

Well, also some other good things like good support, snappy UI, feature-rich, etc. But all of them have it, more or less.

**Comment by mjackson** [comment]
Thanks for the work here, @nkbt, but I'd rather stick with Travis for now. I'm not quite ready to make the jump to Circle CI yet.

**Comment by nkbt** [comment]
No worries, I'll get Travis config ready asap.

**Pull request closed without merging**